documentation/ADRs/009_log_file_for_generated_data.md

```diff
@@ -0,0 +1,49 @@
+# Log file for generated data
+
+
+| ADR Info            | Details                     |
+|---------------------|-----------------------------|
+| Subject             | Log file for generated data |
+| ADR Number          | 009                         |
+| Status              | Accepted                    |
+| Author              | Xiaolong                    |
+| Date                | 09.09.2024                  |
+
+## Context
+In the context of the VIEWS pipeline, there is a need to create a log file to ensure that models and data are tracked accurately and meet certain criteria before running. 
+This log file will provide a detailed record of the data generation process, including the model artifact, the generated data, and the input raw data, to facilitate further checks. 
+This is critical to ensure the reliability and reproducibility of model outputs and to prevent outdated or incorrect data from being used in production systems.
+
+## Decision
+This decision involves implementing a logging system for all generated data and enforce ensemble model checks. 
+This logging will involve creating a **.txt** log file in each model-specific folder. The log file will contain the following details:
+- The name and timestamp of the model artifact that produced the data.
+- The timestamp of when the data was generated.
+- Possibly the data stamp of when the raw data used was fetched from VIEWS.
+- The deployment status of the single model.
+
+Additionally, ensemble models will enforce a set of preconditions before running:
+- The model artifact used must be trained within the current year (after July).
+- The generated data must be from the current month.
+- The raw data must also have been fetched in the current month.
+
```

Comment:
Yes, super. but this sense to use as an example of how it is used in the deployment